arnold-usd fails to compile against USD 25.08

Build failures occur when trying to compile against USD 25.08


```
plugins/node_registry/parser.cpp:99:5: error: 'pxrInternal_v0_25_8__pxrReserved__::SdrSdfTypeIndicator pxrInternal_v0_25_8__pxrReserved__::{anonymous}::ArnoldShaderProperty::GetTypeAsSdfType() const' marked 'override', but does not override
           GetTypeAsSdfType() const override
           ^~~~~~~~~~~~~~~~
plugins/node_registry/parser.cpp:110:20: error: 'const pxrInternal_v0_25_8__pxrReserved__::VtValue& pxrInternal_v0_25_8__pxrReserved__::{anonymous}::ArnoldShaderProperty::GetDefaultValueAsSdfType() const' marked 'override', but does not override
           const VtValue& GetDefaultValueAsSdfType() const override { return _defaultValue; }
```

--- a/plugins/node_registry/parser.cpp
+++ b/plugins/node_registry/parser.cpp
@@ -87,6 +87,9 @@ public:
           _typeName(typeName)
     {
     }
+
+#if PXR_VERSION < 2508
+
 #if PXR_VERSION >= 2505
     SdrSdfTypeIndicator
 #elif PXR_VERSION >= 2411
@@ -109,10 +112,10 @@ public:
 #if PXR_VERSION >= 2111
     const VtValue& GetDefaultValueAsSdfType() const override { return _defaultValue; }
 #endif
+#endif // PXR_VERSION < 2508
 private:
     SdfValueTypeName _typeName;
 };
-
 } // namespace
 
 NdrArnoldParserPlugin::NdrArnoldParserPlugin() {}

--- a/plugins/scene_index/implicitSurfaceSIP.cpp
+++ b/plugins/scene_index/implicitSurfaceSIP.cpp
@@ -180,8 +180,8 @@ HdArnoldImplicitSurfaceSceneIndexPlugin::_AppendSceneIndex(
     const HdSceneIndexBaseRefPtr &inputScene,
     const HdContainerDataSourceHandle &inputArgs)
 {
-#if PXR_VERSION <= 2505
     auto implicitSurfaceSceneIndex = HdsiImplicitSurfaceSceneIndex::New(inputScene, inputArgs);
+#if PXR_VERSION <= 2505
     implicitSurfaceSceneIndex->SetDisplayName("Arnold: implicit surface scene index");
     return _FixImplicitSurfaceSidedNessSceneIndex::New(implicitSurfaceSceneIndex);
 #else // Assuming the sidedness bug is fixed in 25.08

--- a/tools/utils/dependencies.py
+++ b/tools/utils/dependencies.py
@@ -88,7 +88,6 @@ def ndr_plugin(env, sources):
         'tf',
         'gf',
         'vt',
-        'ndr',
         'sdr',
         'sdf',
         'usd',
@@ -99,6 +98,8 @@ def ndr_plugin(env, sources):
     ]
     if env['USD_VERSION_INT'] >= 2411:
         usd_libs += ['boost','python',]
+    if env['USD_VERSION_INT'] < 2508:
+        usd_libs += ['ndr',]
     return add_plugin_deps(env, sources, usd_libs, False)
 
 def usd_imaging_plugin(env, sources):
@@ -111,7 +112,6 @@ def usd_imaging_plugin(env, sources):
         'vt',
         'gf',
         'work',
-        'ndr',
         'sdf',
         'sdr',
         'hf',
@@ -129,6 +129,8 @@ def usd_imaging_plugin(env, sources):
         usd_libs += ['boost','python',]
     if env['USD_VERSION_INT'] >= 2505:
         usd_libs += ['ts',]
+    if env['USD_VERSION_INT'] < 2508:
+        usd_libs += ['ndr',]
     return add_plugin_deps(env, sources, usd_libs, True)
 
 def scene_index_plugin(env, sources):
@@ -141,7 +143,6 @@ def scene_index_plugin(env, sources):
         'vt',
         'gf',
         'work',
-        'ndr',
         'sdf',
         'sdr',
         'hf',
@@ -161,6 +162,8 @@ def scene_index_plugin(env, sources):
         usd_libs += ['boost','python',]
     if env['USD_VERSION_INT'] >= 2505:
         usd_libs += ['ts',]
+    if env['USD_VERSION_INT'] < 2508:
+        usd_libs += ['ndr',]
     return add_plugin_deps(env, sources, usd_libs, True)
 
 def scene_delegate(env, sources):

